docs/tutorials/build-a-module.rst

```diff
@@ -5,5 +5,338 @@
+installed and run the following command:
+
+.. code-block:: console
+
+    $ cookiecutter gh:inveniosoftware/cookiecutter-invenio-module
+    project_name [Invenio-FunGenerator]: Invenio-Foo
+    project_shortname [invenio-foo]:
+    package_name [invenio_foo]:
+    github_repo [inveniosoftware/invenio-foo]:
+    description [Invenio module that adds more fun to the platform.]:
+    author_name [CERN]:
+    author_email [info@inveniosoftware.org]:
+    year [2018]:
+    copyright_holder [CERN]:
+    copyright_by_intergovernmental [True]:
+    superproject [Invenio]:
+    transifex_project [invenio-foo]:
+    extension_class [InvenioFoo]:
+    config_prefix [FOO]:
+
+
+The newly scaffolded module will have the following folder structure::
+
+    invenio-foo/
+        docs/
+        examples/
+        invenio_foo/
+            templates/invenio_foo/
+            __init__.py
+            config.py
+            ext.py
+            version.py
+            views.py
+        tests/
+        *.rst
+        run-tests.sh
+        setup.py
+
+These files are described in the sections below.
+
+\*.rst files
+++++++++++++
+
+All these files are used by people who want to know more about your module (mainly developers).
+
+- ``README.rst`` is used to describe your module. You can see the short
+  description written in the Cookiecutter here. You should update it with
+  more details.
+- ``AUTHORS.rst`` should list all contributors to this module.
+- ``CHANGES.rst`` should be updated at every release and store the list of
+  versions with the list of changes (changelog).
+- ``CONTRIBUTING.rst`` presents the rules to contribute to your module.
+- ``INSTALL.rst`` describes how to install your module.
+
+setup.py
+++++++++
+
+First, there is the ``setup.py`` file, one of the most important: this file is
+executed when you install your module with *pip*. If you open it, you can see
+several parts.
+
+On the top, the list of the requirements:
+
+- For normal use.
+- For development.
+- For tests.
+
+Depending on your needs, you can install only part of the requirements, or
+everything (``pip install invenio-foo[all]``).
+
+Then, in the ``setup()`` function, you can find the description of your module with
+the values entered in cookiecutter. At the end, you can find the
+``entrypoints`` section.
+
+run-tests.sh
+++++++++++++
+This is used to run a list of tests locally, to make sure that your module works
+as intended. It will generate the documentation, run *pytest* and any remaining
+checks.
+
+docs folder
++++++++++++
+This folder contains the settings to generate documentation for your module,
+along with files where you can write the documentation. When you run the
+``run-tests.sh`` script, it will create the documentation in HTML files in a
+sub-folder.
+
+examples folder
++++++++++++++++
+Here you can find a small example of how to use your module. You can test it
+following the steps described in the :ref:`run-the-example-app` section.
```

Comment:
I think we should link to `Run the tests` module instead of `Build the documentation`

Reply:
Thanks, I didn't notice! Actually there the `Run the example application` section :)